src/App.tsx

```diff
@@ -41,6 +59,66 @@ const MainContainer = styled.div`
 
 const App = () => {
   const size: Size = useWindowSize();
+  const dispatch = useDispatch();
+  const currentConfig: ProjectConfig = useSelector(currentConfigSelector);
+  const projects: Project[] = useAppSelector(state => state.config.projects);
+  const projectConfig: ProjectConfig | null | undefined = useAppSelector(state => state.config.projectConfig);
+  const fileMap = useAppSelector(state => state.main.fileMap);
```

Comment:
Since you're only using the root file entry here, this should probably be:
`const rootFileEntry = useAppSelector(state => state.main.fileMap[ROOT_FILE_ENTRY]);`
Because we don't want App.tsx to re-render on every change to `fileMap`, probably only to changes of the root file entry